### DISCUSSION TITLE
Rigid Rotation of the Celestial Sphere and the Implementation of Precession in Stellarium.

### DISCUSSION THREAD

**Discussion by BikramAjay** [Stellarium/stellarium#1831] [General]
Rotation of a rigid body (Rigid Rotation) is defined as where all particles in the body rotate through the same angle during the same time interval. When it comes to the impact of precession on the position of stars, the book titled "Fundamentals of Astronomy" by Barbieri and Bertini (See attached screenshot) states as follows:

"Precession and nutation are rigid rotations of the celestial sphere; as such, they do not alter the angular distance between the stars. The apparent shape of a constellation or of a nebula will change in time, because of the proper motions of stars (described in Chapter 8) and/or expansion of filaments, not because of precession and nutation. However, the position angle p between two objects defined in Chapter 3 will change because it is measured from the variable direction of the north celestial pole."

I have attached the screenshot relevant to the abv para.  Now, if I look at the angular separation between the pair of Merak and Dubhe (from the Big Dipper constellation of Ursa Major), it changes by approx 12 Minutes between the present and 6000 BCE. 

Is the epoch transformation(rotation) implementation in Stellaroim not perfect rigid rotation of celestial sphere? Any pointer to the actual code (filename) is appreciated. 

![9780367253202](https://user-images.githubusercontent.com/86772525/128608258-6012d6b4-e033-4c58-9da0-a80b860041ec.jpg)
![Screenshot 2021-08-06 222203](https://user-images.githubusercontent.com/86772525/128608263-ae125692-cc42-4f17-a6b2-dfd1e5e10475.jpg)
![Screenshot 2021-08-06 222340](https://user-images.githubusercontent.com/86772525/128608264-e9afdbc5-54da-468a-8be3-622f600d6485.jpg)
![Merak_Dubhe_Present](https://user-images.githubusercontent.com/86772525/128608438-a82d7ded-76a1-4936-a7ae-ce483282e588.jpg)
![Merak-Dubhe_6000BC](https://user-images.githubusercontent.com/86772525/128608442-24330d45-2751-4ff6-abc3-f146db1280f1.jpg)

**Comment by gzotti** [comment]
You see the effect of proper motion.

**Reply by BikramAjay** [reply]
I am using the following routine from IAU SOFA to calculate PM and it's not more than approx 1.5 minutes in RA and 4 minutes in dec for either of the stars. Routine uses the linear model (like in Stellarium) for PM calculation. 

http://www.iausofa.org/2021_0512_C/sofa/starpm.c

**Reply by gzotti** [reply]
Please study Stellarium's star code yourself. ZoneArray is a good start to see the stars evaluated, and in utils/Parse*.C you will find code from 2007 which is/was used to create the star catalog. And read Appendix F.2 in the User Guide about better software to find star positions in 6000BC.

**Reply by BikramAjay** [reply]
Just to let you know that even for 800AD, the change in angular separation isn't commensurate with the PM. Thanks for the pointers to the further exploration. 

![Screenshot 2021-08-07 200351](https://user-images.githubusercontent.com/86772525/128617056-3da11e15-7105-48e1-8b94-81ba8f9e1422.jpg)
![Screenshot 2021-08-07 200532](https://user-images.githubusercontent.com/86772525/128617058-85fbe775-b9c5-4aa9-b588-668bb0aeade3.jpg)
